### PR TITLE
Datadog logs

### PR DESCRIPTION
**Pull request by ricardoMogg**
**What changed? Why?**
Add datadog logging for APIs

**Notes to reviewers**
logger object won't work on FE since the datadog key is not present there

**How has it been tested?**
tested locally

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/app/datadog.ts [hunk 1]
```diff
@@ -14,7 +15,7 @@ export default function DatadogInit() {
     if (isDevelopment) return;
 
     if (!nextPublicDatadogAppId || !nextPublicDatadogClientToken) {
-      console.warn('Datadog is not configured');
+      logger.warn('Datadog is not configured');
```
Comment (dneilroth) [inline apps/web/app/datadog.ts:18]: will this actually end up anywhere? STDOUT?

File: apps/web/pages/api/name/[alreadyClaimedName].ts [hunk 2]
```diff
@@ -58,8 +59,8 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse<
     });
     res.status(200).json({ suggestion: JSON.parse(suggestion.response) as string[] });
   } catch (e) {
-    console.error(e);
     if (e instanceof Error) {
+      logger.error(e);
       res.status(500).json({ error: `failed to generate suggestions ${e.message}` });
```
Comment (dneilroth) [inline apps/web/pages/api/name/[alreadyClaimedName].ts:64]: should we stop returning full error messages to FE?

**Review by dneilroth — APPROVED** [review]

File: apps/web/app/datadog.ts (hunk 1 above)
Comment (ricardoMogg) [reply apps/web/app/datadog.ts:18]: it will end up in stdout.

**Pull request merged**